src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,116 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Deadline Class that is a subclass of task, has additional date and time attributes
+ */
+public class Deadline extends Task {
+    private LocalDate date;
+    private LocalTime time = null;
+
+    /**
+     * Constructor for Deadline Class
+     * Input is taken as {DATE TIME}, where date is in YYYY-MM-DD format and time is in HH:mm format
+     *
+     * @param taskName the details/name of the task
+     * @param dateTime the date and time in string format
+     * @throws DukeException checks for any invalid input into date and time
+     */
+    public Deadline(String taskName, String dateTime) throws DukeException {
+        super(taskName);
+        dateTime = dateTime.trim();
+        String[] splitString = dateTime.split(" ");
+        if (splitString.length == 1) {
+            try {
+                this.date = LocalDate.parse(splitString[0]);
+            } catch (Exception e) {
+                throw new DukeException("Invalid input into date");
+            }
+        } else if (splitString.length == 2) {
+            try {
+                this.date = LocalDate.parse(splitString[0]);
+                this.time = LocalTime.parse(splitString[1]);
+            } catch (Exception e) {
+                throw new DukeException("Invalid input into date/time");
+            }
+        } else {
+            throw new DukeException("Please input for a date (optional: time)");
+        }
+    }
+
+    /**
+     * Getter for date
+     *
+     * @return date
+     */
+    public LocalDate getDate() {
+        return date;
+    }
+
+    /**
+     * Getter for time
+     * @return time
```

Comment:
```suggestion
     * 
     * @return time
```

there should be a newline between the description and parameter sections

Reply:
(same for line 97 of this file, line 90 of Event.java, line 32 of Todo.java)